src/cephadm/cephadm

```diff
@@ -3362,7 +3363,13 @@ def _pull_image(ctx, image):
         'Digest did not match, expected',
     ]
 
-    cmd = [ctx.container_engine.path, 'pull', image]
+    cmd = [ctx.container_engine.path, 'pull']
+    if ctx.skip_tls_pull:
+        if ctx.docker:
```

Comment:
```suggestion
        if isinstance(self.ctx.container_engine, Docker):
```
I think the docker flag being checked here only makes us prefer docker over podman. If you have docker installed and don't have podman installed I think it will use docker regardless of whether the docker flag is set, so this check isn't guaranteed to work. Our usual way of checking this is checking if the current container engine instance is a Docker one.